code/modules/station_goals/bsa.dm

```diff
@@ -265,7 +268,12 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/power/bsa/full)
 
 /obj/machinery/power/bsa/full/proc/fire(mob/user, turf/bullseye)
 	winding_up = FALSE
-	playsound(get_turf(src), fire_sound, 50, 1, world.maxx)
+	playsound(get_turf(src), fire_sound, 100, 1, world.maxx, pressure_affected = FALSE, ignore_walls = TRUE)
+	for(var/mob/M in GLOB.mob_living_list)
+		if(M.get_virtual_z_level() != get_virtual_z_level())
+			continue
+		shake_camera(M, 15, 1)
+
```

Comment:
I don't think this can only target station things.
There is `compare_z` proc in pinpointer code. Get a Z group level of its target first, then compare that with people's current z location.

Reply:
done